src/client.rs

```diff
@@ -141,107 +182,123 @@ impl ReconnectingRequest {
     }
 }
 
-impl ReconnectingRequest {
-    fn send_request(&self) -> impl Future<Item = ra::Response, Error = Error> + Send {
-        let request = self
-            .http
-            .get(self.props.url.clone())
-            .headers(self.props.headers.clone());
-        request.send().map_err(|e| Error::HttpRequest(Box::new(e)))
+impl<C> ReconnectingRequest<C> {
+    fn send_request(&self) -> Result<ResponseFuture>
+    where
+        C: Connect + Clone + Send + Sync + 'static,
+    {
+        let mut request = Request::get(&self.props.url);
+        *request.headers_mut().unwrap() = self.props.headers.clone();
+        let request = request
+            .body(Body::empty())
+            .map_err(|e| Error::HttpRequest(Box::new(e)))?;
+        Ok(self.http.request(request))
     }
 
-    fn backoff(&mut self) -> Duration {
+    fn backoff(mut self: Pin<&mut Self>) -> Duration {
```

Comment:
made the receiver `Pin` because these only get called from `poll_next` and this makes some things easier